src/components/ModelingMachineProvider.tsx

```diff
@@ -309,9 +318,16 @@ export const ModelingMachineProvider = ({
               null
             if (!setSelections) return {}
 
-            const dispatchSelection = (selection?: EditorSelection) => {
+            const dispatchSelection = (
+              selection: EditorSelection,
+              scrollIntoView: boolean | undefined
+            ) => {
               if (!selection) return // TODO less of hack for the below please
-              if (!editorManager.editorView) return
+              if (!editorManager.editorView) {
+                console.warn('no editorView')
+                selectionChangedObservable.next(selection)
+                return
+              }
```

Comment:
Good catch here. We do need to be careful. I'm thinking that we may want to wrap this code in a try-catch, where the catch also calls `selectionChangedObservable.next()`. This _must_ get called. If it doesn't, whatever is waiting for it gets stuck.

Reply:
Also, we should maybe clean it up to not send the `selection`. I thought it was a good idea while developing. But we never use it. We can use a [`void` subject](https://rxjs.dev/guide/subject#void-subject) and call simply `selectionChangedObservable.next()` (no argument).

Reply:
Nice find, I've updated